Sync/SyncTelemetryUtils.swift

```diff
@@ -468,6 +468,39 @@ public class GleanSyncOperationHelper {
         GleanMetrics.Pings.shared.tempSync.submit()
     }
 
+    public func reportTelemetry(_ result: MozillaAppServices.SyncResult) {
+        guard let json = result.telemetryJson,
+              let telemetry = try? RustSyncTelemetryPing.fromJSONString(
+                jsonObjectText: json
+              ) else {
+            return
+        }
+
+        for sync in telemetry.syncs {
+            _ = GleanMetrics.Sync.syncUuid.generateAndSet()
+
+            if let failureReason = sync.failureReason {
+                GleanMetrics.Sync.failureReason[String(describing: failureReason)].add()
```

Comment:
@tarikeshaq Basically that `failureReason` isn't reported which is less than ideal. As such I don't think it makes sense to sink any more time into these temp metrics.

I'm going to add some logic to map the new values to those specified in the YAML file as a stopgap to prevent records from being dropped on the floor, push this code to beta so it can be tested, prioritize adding permanent metrics in a-s for iOS, and hold off on the rollout until the permanent metrics are in place.

We can talk about this more tomorrow but just adding this here for posterity.